apis/cloudstack/src/main/java/org/jclouds/cloudstack/compute/options/CloudStackTemplateOptions.java

```diff
@@ -48,265 +55,208 @@
-      this.securityGroupIds.add(securityGroupId);
+   public CloudStackTemplateOptions account(String account) {
+      this.account = account;
       return this;
    }
 
-   /**
-    * @see DeployVirtualMachineOptions#securityGroupIds
-    */
-   public CloudStackTemplateOptions securityGroupIds(Iterable<String> securityGroupIds) {
-      Iterables.addAll(this.securityGroupIds, checkNotNull(securityGroupIds, "securityGroupIds was null"));
+   public CloudStackTemplateOptions domainId(String domainId) {
+      this.domainId = domainId;
       return this;
    }
 
-   public Set<String> getSecurityGroupIds() {
-      return securityGroupIds;
+   public CloudStackTemplateOptions generateKeyPair(boolean generateKeyPair) {
+      this.generateKeyPair = generateKeyPair;
+      return this;
    }
 
-   /**
-    * @see #shouldGenerateKeyPair()
-    */
-   public CloudStackTemplateOptions generateSecurityGroup(boolean enable) {
-      this.generateSecurityGroup = enable;
+   public CloudStackTemplateOptions generateSecurityGroup(boolean generateSecurityGroup) {
+      this.generateSecurityGroup = generateSecurityGroup;
       return this;
    }
 
-   /**
-    * @return true if auto generation of keypairs is enabled
-    */
-   public boolean shouldGenerateSecurityGroup() {
-      return generateSecurityGroup;
+   public CloudStackTemplateOptions diskOfferingId(String diskOfferingId) {
+      this.diskOfferingId = diskOfferingId;
+      return this;
    }
 
-   /**
-    * @deprecated See TemplateOptions#networks
-    * @see DeployVirtualMachineOptions#networkId
-    */
-   @Deprecated
-   public CloudStackTemplateOptions networkId(String networkId) {
-      this.networks.add(networkId);
+   public CloudStackTemplateOptions dataDiskSize(int dataDiskSize) {
+      this.dataDiskSize = dataDiskSize;
       return this;
    }
 
-   /**
-    * @deprecated See TemplateOptions#networks
-    * @see DeployVirtualMachineOptions#networkIds
-    */
-   @Deprecated
-   public CloudStackTemplateOptions networkIds(Iterable<String> networkIds) {
-      Iterables.addAll(this.networks, checkNotNull(networkIds, "networkIds was null"));
+   public CloudStackTemplateOptions userData(byte[] userData) {
+      this.userData = userData;
       return this;
    }
 
-   /**
-    * @deprecated See TemplateOptions#getNetworks
-    */
-   @Deprecated
-   public Set<String> getNetworkIds() {
-      return this.getNetworks();
+   public CloudStackTemplateOptions userData(String userData) {
+      this.userData = checkNotNull(userData, "userdata").getBytes(Charsets.UTF_8);
+      return this;
    }
 
-   public CloudStackTemplateOptions setupStaticNat(boolean setupStaticNat) {
-      this.setupStaticNat = setupStaticNat;
+   public CloudStackTemplateOptions userData(URL userDataUrl) throws IOException {
+      this.userData = Resources.toString(checkNotNull(userDataUrl, "userDataUrl"), Charsets.UTF_8).getBytes(Charsets.UTF_8);
```

Comment:
Why providing this method that accepts a URL? I don't see it necessary and only introduces issues: can we guarantee the classloader being used to load the resource the right one to load the file? And if jclouds code is deployed in a webapp? Where do we expect to load files from?

IMO there is no need to provide this "sugar" method. Let users read files/resources themselves and just pass the array of bytes.